### PR TITLE
Rename partitioner functions

### PR DESCRIPTION
**Pull request by ddn0**
This is my last tidying PR before I replace ArrowIterator.h with Properties.h. It was probably ill-advised, but I think after this, cleaning up how properties are loaded (the next thing on my plate) and untangling the Distribution class, the Partitioner should be pretty maintainable and readable.

The following is from the main commit in this PR:

There is a some inconsistency in the naming of the internal partitioner
functions. "sync" can mean synchronize, receive or synchronous. "async"
means asynchronous but it sometimes appears at the beginning functions
where other functions begin with "sync" and the latter means
synchronize. Major phases are sometimes numbered (e.g., "phase0") and
sometimes do not have a specific function associated with them (e.g.,
everything else).

Instead,
- start using DoFoo for the major phases of partitioning
- use SendRecvFoo for synchronizing Foo
- use FooAsync for asynchronous functions

Along the way, update code style of any touched functions.

I left the property send and receive functions as-is because I'll be
refactoring those in a future commit.

### CODE REVIEW

File: libcusp/include/galois/graphs/Partitioner.h [hunk 1]
```diff
@@ -285,40 +282,40 @@ class Partitioner {
...
-
-    // Step 4: send bitset to other hosts
-    for (unsigned h = 0; h < dist_.num_hosts_; h++) {
-      galois::runtime::SendBuffer bitsetBuffer;
-
-      if (h != dist_.host_) {
-        galois::runtime::gSerialize(bitsetBuffer, syncNodes[h]);
-        bytesSent += bitsetBuffer.size();
-        net.sendTagged(h, galois::runtime::evilPhase, bitsetBuffer);
+    galois::StatTimer timer("Phase0SendRecvBitsets", kGRName);
```
Comment (gurbinder533) [inline libcusp/include/galois/graphs/Partitioner.h:288]: Remove phase0 from the timer name as well.

File: libcusp/include/galois/graphs/Partitioner.h [hunk 2]
```diff
@@ -377,34 +374,34 @@ class Partitioner {
...
-  void asyncSendLoad(galois::PODResizeableArray<uint64_t>& nodeAccum,
-                     galois::PODResizeableArray<uint64_t>& edgeAccum) {
+  void SendLoadsAsync(PODResizeableArray<uint64_t>& node_accum,
+                      PODResizeableArray<uint64_t>& edge_accum) {
     auto& net = galois::runtime::getSystemNetworkInterface();
 
-    unsigned bytesSent = 0;
-    galois::StatTimer sendTimer("Phase0AsyncSendLoadTime", kGRName);
+    unsigned bytes_sent = 0;
+    galois::StatTimer timer("Phase0AsyncSendLoadTime", kGRName);
```
Comment (gurbinder533) [inline libcusp/include/galois/graphs/Partitioner.h:382]: Same.

File: libcusp/include/galois/graphs/Partitioner.h [hunk 3]
```diff
@@ -181,43 +179,42 @@ class Partitioner {
    *     me)
    */
   // steps 1 and 2 of neighbor location setup: memory allocation, bitset setting
-  void phase0BitsetSetup(const PartialV1GraphView& pview,
-                         galois::DynamicBitSet& ghosts) {
-    galois::StatTimer bitsetSetupTimer("Phase0BitsetSetup", kGRName);
-    bitsetSetupTimer.start();
+  void DoMasterAssignmentBitsetSetup(const PartialV1GraphView& pview,
+                                     DynamicBitSet& ghosts) {
+    galois::StatTimer timer("Phase0BitsetSetup", kGRName);
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:184]: Function name as timer name

File: libcusp/include/galois/graphs/Partitioner.h [hunk 4]
```diff
@@ -229,53 +226,53 @@ class Partitioner {
...
-    mapSetupTimer.start();
-
-    uint32_t numLocal = dist_.gid_to_host_[dist_.host_].second -
-                        dist_.gid_to_host_[dist_.host_].first;
-    uint32_t lid = numLocal;
+  uint64_t DoMasterAssignmentMapSetup(
+      DynamicBitSet& ghosts,
+      std::unordered_map<uint64_t, uint32_t>& gid_to_offsets,
+      galois::gstl::Vector<galois::gstl::Vector<uint32_t>>& sync_nodes) {
+    galois::StatTimer timer("Phase0MapSetup", kGRName);
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:233]: Ditto

File: libcusp/include/galois/graphs/Partitioner.h [hunk 5]
```diff
@@ -377,34 +374,34 @@ class Partitioner {
...
         // note the +1 on evil phase; load messages send using a different
         // phase to avoid conflicts
-        net.sendTagged(h, dist_.evilPhasePlus1(), b);
+        net.sendTagged(host, dist_.evilPhasePlus1(), b);
       }
     }
-    sendTimer.stop();
+    timer.stop();
 
     galois::runtime::reportStat_Tsum(kGRName, "Phase0AsyncSendLoadBytesSent",
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:403]: Ditto

File: libcusp/include/galois/graphs/Partitioner.h [hunk 6]
```diff
@@ -573,64 +571,62 @@ class Partitioner {
...
-    auto& net              = galois::runtime::getSystemNetworkInterface();
-    std::string statString = std::string("Phase0SendOffsets_") + timerName;
-    uint64_t bytesSent     = 0;
+  void SendOffsets(unsigned target_host, DynamicBitSet& to_send,
+                   std::vector<uint32_t>& data_vector) {
+    auto& net           = galois::runtime::getSystemNetworkInterface();
+    uint64_t bytes_sent = 0;
 
-    galois::StatTimer sendOffsetsTimer(statString.c_str(), kGRName);
+    galois::StatTimer timer("Phase0SendOffsets", kGRName);
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:579]: Ditto

File: libcusp/include/galois/graphs/Partitioner.h [hunk 7]
```diff
@@ -573,64 +571,62 @@ class Partitioner {
...
-      net.sendTagged(targetHost, galois::runtime::evilPhase, b);
+      bytes_sent += b.size();
+      net.sendTagged(target_host, galois::runtime::evilPhase, b);
     }
-    sendOffsetsTimer.stop();
+    timer.stop();
 
-    galois::runtime::reportStat_Tsum(kGRName, statString + "BytesSent",
-                                     bytesSent);
+    galois::runtime::reportStat_Tsum(kGRName, "Phase0SendOffsetsBytesSent",
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:628]: Ditto

File: libcusp/include/galois/graphs/Partitioner.h [hunk 8]
```diff
@@ -644,90 +640,90 @@ class Partitioner {
...
-      uint32_t begin, uint32_t end, uint32_t numLocalNodes,
-      std::vector<uint32_t>& localNodeToMaster,
-      galois::gstl::Vector<galois::gstl::Vector<uint32_t>>& syncNodes) {
-    galois::StatTimer p0assignSendTime("Phase0AssignmentSendTime", kGRName);
-    p0assignSendTime.start();
+  void SendMasterAssignments(
+      uint32_t begin, uint32_t end, uint32_t num_local_nodes,
+      std::vector<uint32_t>& local_node_to_master,
+      galois::gstl::Vector<galois::gstl::Vector<uint32_t>>& sync_nodes) {
+    galois::StatTimer timer("Phase0AssignmentSendTime", kGRName);
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:647]: Ditto

File: libcusp/include/galois/graphs/Partitioner.h [hunk 9]
```diff
@@ -836,48 +834,45 @@ class Partitioner {
...
-  void
-  syncAssignmentReceives(std::vector<uint32_t>& localNodeToMaster,
-                         std::unordered_map<uint64_t, uint32_t>& gid2offsets) {
-    galois::StatTimer p0assignReceiveTime("Phase0AssignmentReceiveTime",
-                                          kGRName);
-    p0assignReceiveTime.start();
+  void RecvMasterAssignments(
+      std::vector<uint32_t>& local_node_to_master,
+      std::unordered_map<uint64_t, uint32_t>& gid_to_offsets) {
+    galois::StatTimer timer("Phase0AssignmentReceiveTime", kGRName);
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:845]: Ditto

File: libcusp/include/galois/graphs/Partitioner.h [hunk 10]
```diff
@@ -836,48 +834,45 @@ class Partitioner {
...
-      std::unordered_map<uint64_t, uint32_t>& gid2offsets,
-      galois::DynamicBitSet& hostFinished) {
-    galois::StatTimer p0assignReceiveTime("Phase0AssignmentReceiveTimeAsync",
-                                          kGRName);
-    p0assignReceiveTime.start();
+  void RecvMasterAssignmentsAsync(
+      std::vector<uint32_t>& local_node_to_master,
+      std::unordered_map<uint64_t, uint32_t>& gid_to_offsets,
+      DynamicBitSet& host_finished) {
+    galois::StatTimer timer("Phase0AssignmentReceiveTimeAsync", kGRName);
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:869]: Ditto

File: libcusp/include/galois/graphs/Partitioner.h [hunk 11]
```diff
@@ -1335,22 +1297,22 @@ class Partitioner {
   }
 
   /**
-   * Given a loaded graph, construct the edges in the DistGraph graph.
-   * Variant that constructs edge data as well.
+   * DistributeEdgeCut constructs edges assuming the graph is already partially
+   * constructed by DoEdgeAssignment.
```
Comment (roshandathathri) [inline libcusp/include/galois/graphs/Partitioner.h:1301]: the graph is already allocated and edge-indices is already constructed based on DoEdgeAssignment

**Pull request merged**